packages/yoga/src/AutoComplete/web/AutoComplete.jsx

```diff
@@ -159,14 +159,19 @@ const AutoComplete = ({
     <Downshift
       selectedItem={userValue}
       onStateChange={changes => {
-        if (Object.prototype.hasOwnProperty.call(changes, 'selectedItem')) {
-          setUserValue(changes.selectedItem);
-          onSelect(changes.selectedItem);
+        const { selectedItem, inputValue } = changes;
+        if (
+          Object.prototype.hasOwnProperty.call(changes, 'selectedItem') &&
+          selectedItem
```

Comment:
Is there a difference between `Object.prototype.hasOwnProperty.call(changes, 'selectedItem')` and `Boolean(changes.selectedItem)` in this case? Same applies to the `inputValue` validation.

What would happen if we just used then directly?

```js
if (selectedItem) {
  /* ... */
} else if (inputValue) {
  /* ... */
}
```

Reply:
When a user types or selects some suggestion on the field, `changes` object will have "selectedItem" and/or "inputValue" keys inside it.

Talking about "selectedItem" it doesn't make any difference using hasOwnProperty or not.

When we talk about `inputValue` we must call `onChange` even if the "inputValue" is an empty string (like if the user clicks on the "clean" button or deletes the whole string). So, we need to make sure that "inputValue" key exists (even with a falsy value).

Reply:
That's perfect.